spacy_cld/spacy_cld.py

```diff
@@ -17,7 +17,10 @@ def get_scores(text, cld_results=None):
 
 
 def detect_languages(text):
-    _, _, results = detect(text.text)
+    try:
+        _, _, results = detect(text.text)
+    except:
+        results = [[None, "error", 0.0, None]]
```

Comment:
Another suggestion - it is actually often easier for end users if a function explicitly throws an error instead of trying to squash errors. For example, some users might want to exclude any text that raises parsing errors from pieces further down the pipeline. In that case, it's usually easier for them to wrap any calls to `spacy_cld` with a `try...except` block in their code instead of parsing the outputs from `spacy_cld` to determine whether an error was thrown.

For this reason, my suggestion would be to define a custom error class in `spacy_cld/spacy_cld.py` that inherits from the base Python `Exception` class:

```python
class SpacyCldError(Exception):
    pass
```

Then, instead of putting an error key in the `results` object, you can `raise SpacyCldError from err`. This makes it very clear for users who want to catch any errors coming from this module.

I'd also suggest `import`ing `SpacyCldError` in `__init__.py` so that users can easily `import` that error into their code and catch it.

Reply:
I agree with you that throwing exceptions is often beneficial, but in this case an exception will crash the Spacy processing pipeline. So in the end the user will loose access to all docs processed in the current batch (usually 1000s) of the pipeline and still won't know which sentence caused the issue.